### PR TITLE
[3.x]Upgrade Slurm to 22.05.3

### PR DESCRIPTION
**Pull request by hanwen-pcluste**
In addition to the upgrade:
1. Remove TaskAffinity=no in cgroups.conf. This parameter had been ignored from Slurm 21. In addition, Slurm 22 fails if the parameter still exists.
2. Remove prolog/epilog used as a workaround for cluster without Internet connection.

Signed-off-by: Hanwen <hanwenli@amazon.com>

### Tests
Integration tests in `configs/pcluster3.yaml` and `test_cluster_in_no_internet_subnet` have been passed

### References
This PR should be merged together with https://github.com/aws/aws-parallelcluster/pull/4347

### Checklist
- [x] Make sure you are pointing to **the right branch** and add a label in the PR title (i.e. **2.x** vs **3.x**)
- [x] Check all commits' messages are clear, describing what and why vs how.
- [x] Make sure **to have added unit tests or integration tests** to cover the new/modified code.
- [ ] Check if documentation is impacted by this change.

Please review the [guidelines for contributing](../CONTRIBUTING.md) and [Pull Request Instructions](https://github.com/aws/aws-parallelcluster/wiki/Git-Pull-Request-Instructions).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: cookbooks/aws-parallelcluster-slurm/recipes/config_head_node.rb [hunk 1]
```diff
@@ -192,34 +192,6 @@
...
 end
 
-if node['cluster']['add_node_hostnames_in_hosts_file'] == "true"
-  directory "#{node['cluster']['slurm']['install_dir']}/etc/pcluster/prolog.d" do
-    user 'root'
-    group 'root'
-    mode '0755'
-  end
-
-  cookbook_file "#{node['cluster']['slurm']['install_dir']}/etc/pcluster/prolog.d/01-pcluster-prolog" do
```
Comment (demartinofra) [inline cookbooks/aws-parallelcluster-slurm/recipes/config_head_node.rb:202]: you might also want to delete the content of this file

File: cookbooks/aws-parallelcluster-slurm/recipes/config_head_node.rb [hunk 2]
```diff
@@ -192,34 +192,6 @@
...
-    mode '0744'
-  end
-
-  directory "#{node['cluster']['slurm']['install_dir']}/etc/pcluster/epilog.d" do
-    user 'root'
-    group 'root'
-    mode '0755'
-  end
-
-  cookbook_file "#{node['cluster']['slurm']['install_dir']}/etc/pcluster/epilog.d/01-pcluster-epilog" do
```
Comment (demartinofra) [inline cookbooks/aws-parallelcluster-slurm/recipes/config_head_node.rb:215]: same here

File: cookbooks/aws-parallelcluster-slurm/templates/default/slurm/slurm.conf.erb [hunk 3]
```diff
@@ -40,19 +40,6 @@ SuspendTimeout=120
...
-# prolog is executed to add nodes info to /etc/hosts on compute nodes when each job is allocated
-# epilog is executed to clean contents written by prolog
-# PrologFlags specifies the prolog is executed at job allocation and prologs and epilogs are of different jobs are executed serially
-# SchedulerParameters allows jobs to be requeued to other nodes if prolog error exits.
-# Note the error exit of prolog drains a node, because the error of prolog is considered as a node error.
-Epilog=<%= node['cluster']['slurm']['install_dir'] %>/etc/pcluster/epilog.d/*
-Prolog=<%= node['cluster']['slurm']['install_dir'] %>/etc/pcluster/prolog.d/*
-PrologFlags=alloc,serial
-SchedulerParameters=nohold_on_prolog_fail
-<% end -%>
```
Comment (jdeamicis) [inline cookbooks/aws-parallelcluster-slurm/templates/default/slurm/slurm.conf.erb:55]: If the decision to remove this is independent of the Slurm upgrade, I would split the commit in two (we can keep the commits in the same PR if you wish).

File: cookbooks/aws-parallelcluster-slurm/recipes/config_head_node.rb (hunk 1 above)
Comment (hanwen-pcluste) [reply cookbooks/aws-parallelcluster-slurm/recipes/config_head_node.rb:202]: Done

File: cookbooks/aws-parallelcluster-slurm/recipes/config_head_node.rb (hunk 2 above)
Comment (hanwen-pcluste) [reply cookbooks/aws-parallelcluster-slurm/recipes/config_head_node.rb:215]: Done

File: cookbooks/aws-parallelcluster-slurm/templates/default/slurm/slurm.conf.erb (hunk 3 above)
Comment (hanwen-pcluste) [reply cookbooks/aws-parallelcluster-slurm/templates/default/slurm/slurm.conf.erb:55]: Done

**Review by demartinofra — APPROVED** [review]
I'm approving this but let's make sure we run some additional manual checks on scheduler and scaling logs

**Review by jdeamicis — APPROVED** [review]

**Pull request merged**